react/src/components/AssetDetail/AssetDetail.tsx

```diff
@@ -0,0 +1,172 @@
+import React, { Suspense, useState, useEffect } from 'react';
+import _ from 'lodash';
+import { useAppConfiguration, useFeatureSelection } from '@hazmapper/hooks';
+import { useLocation, useNavigate } from 'react-router-dom';
+import { Asset, FeatureTypeNullable } from '@hazmapper/types';
+import { FeatureIcon } from '@hazmapper/components/FeatureIcon';
+import { Button, LoadingSpinner, SectionMessage } from '@tacc/core-components';
+import styles from './AssetDetail.module.css';
+import { faCameraRetro, faPhotoFilm } from '@fortawesome/free-solid-svg-icons';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+
+type AssetModalProps = {
+  isOpen: boolean;
+  projectId: number;
+  isPublicView: boolean;
+};
+
+const AssetDetail: React.FC<AssetModalProps> = ({
+  isOpen,
+  projectId,
+  isPublicView,
+}) => {
+  const { selectedFeature, selectedFeatureId, setSelectedFeatureId } =
+    useFeatureSelection();
+
+  const navigate = useNavigate();
+  const location = useLocation();
+
+  const closePanel = () => {
+    const params = new URLSearchParams(location.search);
+    params.delete('selectedFeature');
+    navigate(`${location.pathname}?${params.toString()}`);
+  };
+
+  const config = useAppConfiguration();
+  const geoapiUrl = config.geoapiUrl;
+
+  const [selectedFeatureAsset, setSelectedFeatureAsset] = useState<
+    Asset | undefined
+  >(selectedFeature?.assets[0]);
+
+  useEffect(() => {
+    const featureAsset = selectedFeature?.assets[0];
+    setSelectedFeatureAsset(featureAsset);
+  }, [selectedFeature]);
+
+  const featureSource: string | undefined =
+    geoapiUrl + '/assets/' + selectedFeatureAsset?.path;
+
+  const fileType: string | undefined = selectedFeatureAsset?.asset_type;
+
+  const AssetRenderer = React.memo(
+    ({
+      type,
+      source,
```

Comment:
I think it would be easier to have one parameter Feature.

as `source` varies with FeatureType.  and even download button and other things vary.  this is just a thought but could AssetRender contain the preview and action button (Download, View) ?

Reply:
Would it be ok to circle back to this when I work on [Asset Information Panel - Support Pointclouds](https://tacc-main.atlassian.net/browse/WG-393) and [Asset Information Panel - Support Questionnaire](https://tacc-main.atlassian.net/browse/WG-389)? It might be easier for me to sort out the best way for the AssetRenderer to work when there's more than just placeholders there. I can add this to both those tasks if that's ok with you?

Reply:
good idea 👍